network/shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -163,12 +163,12 @@ public void registerExecutor(
 
   /**
    * Obtains a FileSegmentManagedBuffer from a shuffle block id. We expect the blockId has the
-   * format "shuffle_ShuffleId_MapId_ReduceId" (from ShuffleBlockId), and additionally make
-   * assumptions about how the hash and sort based shuffles store their data.
+   * format "shuffle_ShuffleId_MapId_ReduceId_StageAttemptId" (from ShuffleBlockId), and
+   * additionally make assumptions about how the hash and sort based shuffles store their data.
    */
   public ManagedBuffer getBlockData(String appId, String execId, String blockId) {
     String[] blockIdParts = blockId.split("_");
-    if (blockIdParts.length < 4) {
+    if (blockIdParts.length != 5) {
```

Comment:
Hmm... is the contract that a new shuffle service should work with apps running an old Spark version? In that case, shouldn't this have different code paths for `length == 4` and `length == 5`?